compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10693,13 +10630,10 @@ private STNode parseOrderKeyList(boolean isRhsExpr) {
 
         // Parse the remaining order keys
         nextToken = peek();
-        STNode orderKeyListMemberEnd;
+        STNode leadingComma;
         while (!isEndOfOrderKeys(nextToken.kind)) {
-            orderKeyListMemberEnd = parseOrderKeyListMemberEnd();
-            if (orderKeyListMemberEnd == null) {
-                break;
-            }
```

Comment:
Wouldn't we need this?
For eg: In case when theres an additional after the order-keys, but not one of the "EndOfOrderKeys" tokens ..?

Reply:
```java    
private boolean isEndOfOrderKeys(SyntaxKind tokenKind) {
        switch (tokenKind) {
            case COMMA_TOKEN:
            case ASCENDING_KEYWORD:
            case DESCENDING_KEYWORD:
                return false;
            default:
                return !isTypeStartingToken(tokenKind);
        }
    }
```
If there's an additional token and not one of end of order keys it should be one of following.
* comma token - recover with missing identifer
* ascending/ descending token - recover with missing ident 
* type start token - it is considered as an expr

Can there be any other case?

Reply:
I think not all type-starting-tokens are expressions..
I was thinking, we can terminate if the next token is an end of order keys for sure. If not, try to parse the "parseOrderKeyListMemberEnd" and let the error recovery handle whether to terminate or continue.. WDYT?

Reply:
yea. that's better. Fixed in a434d8b